manifests/portstree.pp

```diff
@@ -3,13 +3,21 @@
 # parameter
 
 define poudriere::portstree (
-  $ensure           = 'present',
-  $portstree        = $name,
-  $branch           = undef,
-  $fetch_method     = 'svn',
-  $cron_enable      = false,
-  $cron_always_mail = false,
-  $cron_interval    = { minute => 0, hour => 22, monthday => '*', month => '*', week => '*' },
+  Enump['present', 'absent'] $ensure           = 'present',
+  String[1]                  $portstree        = $name,
+  Optional[String[1]]        $branch           = undef,
+  Enum[
+    'git',
+    'null',
+    'portsnap',
+    'svn',
+    'svn+file',
+    'svn+http',
+    'svn+https',
+  'svn+ssh']  $fetch_method     = 'svn',
```

Comment:
this enum could be moved into its own type definition

Reply:
It's used a single time to I was not happy with it, but the horror required by the buggy puppet-strings to cope with this make me think we definitively SHALL do it…

Reply:
Fixed!